src/celeritas/io/EventReader.hh

```diff
@@ -22,18 +24,19 @@ struct Primary;
 
 //---------------------------------------------------------------------------//
 /*!
- * Read an event record file using the HepMC3 event record library and create
- * primary particles.
+ * Read a HepMC3 event record file and create primary particles.
  *
- * Supported forrmats are Asciiv3, IO_GenEvent, HEPEVT, and LHEF.
+ * Each \c operator() call returns a vector of primaries from a single event
+ * until all events have been read. Supported forrmats are Asciiv3,
+ * IO_GenEvent, HEPEVT, and LHEF.
  */
 class EventReader
 {
   public:
     //!@{
     //! \name Type aliases
     using SPConstParticles = std::shared_ptr<const ParticleParams>;
-    using result_type      = std::vector<Primary>;
+    using VecPrimary       = std::vector<Primary>;
```

Comment:
It might be good to leave this as `result_type`: it can be useful to have the same function-like interface as `std::function` (though in this case I can't really see it mattering?)

Reply:
Sure, I can switch it back.